Shadowblade_main.lua

```diff
@@ -0,0 +1,330 @@
+-- takes care at load/reload, <name_#> to be replaced by some convinient name.
+    local unitSelf = self.core.unitSelf
+	
+    if not bSkillsValid then
+        skills.abilGargantuan 	  = unitSelf:GetAbility(0)
+        skills.abilFeint	  = unitSelf:GetAbility(1)
+        skills.abilSoul	  = unitSelf:GetAbility(2)
+        skills.abilEssence 		  = unitSelf:GetAbility(3)
+		skills.abilAttributeBoost = unitSelf:GetAbility(4)
+
+		if skills.abilGargantuan and skills.abilFeint and skills.abilSoul and skills.abilEssence and skills.abilAttributeBoost then
+			bSkillsValid = true	
+		else 
+			return
+		end
+	end
+    
+    if unitSelf:GetAbilityPointsAvailable() <= 0 then
+        return
+    end
+    
+    local nlev = unitSelf:GetLevel()
+    local nlevpts = unitSelf:GetAbilityPointsAvailable()
+    for i = nlev, nlev+nlevpts do
+        unitSelf:GetAbility( object.tSkills[i] ):LevelUp()
+    end
+end
+
+----------------------------------
+--	Shadowblade items
+----------------------------------
+behaviorLib.StartingItems = 
+	{"Item_LoggersHatchet", "Item_IronBuckler", "Item_ManaPotion", "Item_RunesOfTheBlight"}
+behaviorLib.LaneItems = 
+	{"Item_Marchers", "Item_Steamboots"} -- Items: Marchers, Upg Marchers to Steamboots
+behaviorLib.MidItems = 
+	{"Item_Lightbrand", "Item_Sicarius", "Item_Strength6", "Item_Intelligence7"} -- Items: Build Dawnbringer, Staff Of The Master
+behaviorLib.LateItems = 
+	{"Item_Evasion", "Item_BehemothsHeart", 'Item_Damage9'} -- Items: Wingbow, Behemoth's Heart, DoomBringer
+
+
+---------------------------------------------------
+--                   Overrides                   --
+---------------------------------------------------
+
+--[[for testing
+function object:onthinkOverride(tGameVariables)
+	self:onthinkOld(tGameVariables)
+	
+	core.unitSelf:TeamShare()
+	
+	-- Insert code here
+end
+object.onthinkOld = object.onthink
+object.onthink 	= object.onthinkOverride
+--]]
+
+----------------------------------
+--	Shadowblade specific harass bonuses
+--
+--  Abilities off cd increase harass util
+--  Ability use increases harass util for a time
+----------------------------------
+
+object.nGargantuanUp = 20
+object.nFeintUp = 20	
+object.nSoulUp = 20
+
+
+object.nGargantuanUseBonus = 16
+object.nFeintUseBonus = 12
+object.nSoulUseBonus = 10
+
+
+--ability use gives bonus to harass util for a while
+function object:oncombateventOverride(EventData)
+	self:oncombateventOld(EventData)
+	
+	local nAddBonus = 0
+	if EventData.Type == "Ability" then
+			
+		if EventData.InflictorName == "Ability_ShadowBlade1" then
+			nAddBonus = nAddBonus + object.nGargantuanUseBonus
+		end
+		if EventData.InflictorName == "Ability_ShadowBlade2" then
+			nAddBonus = nAddBonus + object.nFeintUseBonus
+		end
+		if EventData.InflictorName == "Ability_ShadowBlade3" then
+			nAddBonus = nAddBonus + object.nSoulUseBonus
+		end
+		if EventData.InflictorName == "Ability_ShadowBlade4" then
+			nAddBonus = nAddBonus + object.nSoulUseBonus
```

Comment:
Using nSoulUseBonus here is a little redundant: SB's ultimate is currently never used in the harass actions and is only present in the retreat function, so you don't need it added to your bonus. I would also suggest adding a UseBonus var for Essence Shift if you do keep this snippet, just for good form.